qiskit_experiments/library/randomized_benchmarking/clifford_utils.py

```diff
@@ -569,6 +570,20 @@ def _create_cliff_2q_layer_2():
 _NUM_LAYER_1 = 20
 _NUM_LAYER_2 = 16
 
+# Construct a dense multiplication table
+_CLIFFORD_LAYER_NUMS=[]
+for layer in [0,1,2]:
+    _CLIFFORD_LAYER_NUMS.append([])
+    for idx, qc in enumerate(_CLIFFORD_LAYER[layer]):
+        nn = []
+        for inst in qc:
+            qubits = tuple(qc.find_bit(q).index for q in inst.qubits)
+            rhs = _num_from_2q_gate(op=inst.operation, qubits=qubits)
+            nn.append(rhs)
+        _CLIFFORD_LAYER_NUMS[layer].append(tuple(nn))
```

Comment:
I understand your updates implies `_compose_num_with_circuit_2q` is not well-designed. So how about replacing `_compose_num_with_circuit_2q` with a new function
```
def _clifford_2q_nums_from_2q_circuit(qc: QuantumCircuit) -> Iterable[Integral]:
    """Yield Clifford numbers that represents the 2Q Clifford circuit."""
    for inst in qc:
        qubits = tuple(qc.find_bit(q).index for q in inst.qubits)
        yield _num_from_2q_gate(op=inst.operation, qubits=qubits)
```
and updating `num_from_2q_circuit` to use the new one?
Using the new function, we can build _CLIFFORD_LAYER_NUMS much cleaner:
```
_CLIFFORD_LAYER_NUMS = [
    [tuple(_clifford_2q_nums_from_2q_circuit(qc)) for qc in _CLIFFORD_LAYER[layer]]
    for layer in [0, 1, 2]
]
```